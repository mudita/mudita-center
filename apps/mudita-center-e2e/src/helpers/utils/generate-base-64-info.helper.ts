/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import crc32 from "crc-32"

export function generateBase64Info(obj: Record<string, unknown>) {
  const jsonString = JSON.stringify(obj)
  const data = Buffer.from(jsonString, "utf8").toString()
  const base64String = Buffer.from(data).toString("base64")
  const sizeInBytes = Buffer.byteLength(data, "utf8")
  const crc32Hex = (crc32.buf(Buffer.from(base64String, "utf8")) >>> 0)
    .toString(16)
    .padStart(8, "0")

  return {
    base64: base64String,
    sizeInBytes: sizeInBytes,
    crc32Hex: crc32Hex,
  }
}
