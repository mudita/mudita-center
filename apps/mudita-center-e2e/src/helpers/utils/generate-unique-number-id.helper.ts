/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

export const generateUniqueNumber = () =>
  Date.now() + Math.floor(Math.random() * 1000)
