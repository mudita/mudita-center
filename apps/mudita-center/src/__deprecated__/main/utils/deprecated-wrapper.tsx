/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { FunctionComponent } from "react"

// AUTO DISABLED - fix me if you like :)
// eslint-disable-next-line @typescript-eslint/no-explicit-any
type Element = FunctionComponent<any>

const deprecated = (component: Element, message?: string): Element => {
  if (process.env.NODE_ENV === "development") {
    const basicMessage = `This component is deprecated.`
    if (message) {
      console.warn(`${basicMessage} ${message}`)
    } else {
      console.warn(basicMessage)
    }
  }

  return component
}

export default deprecated
