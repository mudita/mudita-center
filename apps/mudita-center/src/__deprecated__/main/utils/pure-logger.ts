/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { ConsoleLogger } from "App/core/types"
import logger from "App/__deprecated__/main/utils/logger"
import { flags, Feature } from "App/feature-flags"

export interface ScrubProps {
  // AUTO DISABLED - fix me if you like :)
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  body?: any
  endpoint: number
  status: number
  uuid: number
}

class PureLogger implements ConsoleLogger {
  info(message: string): void {
    if (flags.get(Feature.LogsScrubbingEnabled)) {
      logger.info(this.scrub(message))
    } else {
      logger.info(this.scrubFileSystemData(message))
    }
  }

  private scrub(message: string): string {
    try {
      const messageParsed = JSON.parse(message) as ScrubProps[]
      if (messageParsed[0].body) {
        return JSON.stringify(
          [{ ...messageParsed[0], body: "scrubbed" }],
          null,
          2
        )
      } else {
        return message
      }
    } catch {
      return message
    }
  }

  private scrubFileSystemData(message: string): string {
    try {
      const messageParsed = JSON.parse(message) as ScrubProps[]
      if (
        messageParsed[0].body &&
        messageParsed[0].endpoint === 3 &&
        // AUTO DISABLED - fix me if you like :)
        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
        Boolean(messageParsed[0].body?.data)
      ) {
        return JSON.stringify(
          [
            {
              ...messageParsed[0],
              // AUTO DISABLED - fix me if you like :)
              // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
              body: { ...messageParsed[0].body, data: "scrubbed" },
            },
          ],
          null,
          2
        )
      } else {
        return message
      }
    } catch {
      return message
    }
  }
}

export default PureLogger
