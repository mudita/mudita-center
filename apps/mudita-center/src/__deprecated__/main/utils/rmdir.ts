/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import fs, { RmDirOptions } from "fs"

export interface RmdirProps {
  cwd: string
  options: RmDirOptions
}

const rmdir = ({ cwd, options }: RmdirProps): boolean => {
  try {
    if (fs.existsSync(cwd)) {
      fs.rmdirSync(cwd, options)
    }
    return true
  } catch {
    return false
  }
}

export default rmdir
