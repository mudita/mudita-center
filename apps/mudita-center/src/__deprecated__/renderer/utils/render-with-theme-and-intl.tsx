/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "App/__deprecated__/renderer/locales/default/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "App/__deprecated__/renderer/utils/extract-test-locale"
import ReactModal from "react-modal"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

// AUTO DISABLED - fix me if you like :)
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function constructWrapper(ui: React.ReactElement) {
  return (
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>
  )
}

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    constructWrapper(ui),
    // AUTO DISABLED - fix me if you like :)
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    options
  )
}
