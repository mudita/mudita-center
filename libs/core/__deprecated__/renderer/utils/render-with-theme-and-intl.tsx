/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import ReactModal from "react-modal"
import { ThemeProvider } from "styled-components"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "Core/__deprecated__/renderer/locales/default/en-US.json"
import extractLanguageKeys from "Core/__deprecated__/renderer/utils/extract-test-locale"
import theme from "Core/core/styles/theming/theme"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

// AUTO DISABLED - fix me if you like :)
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function constructWrapper(ui: React.ReactElement) {
  return (
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>
  )
}

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    constructWrapper(ui),
    // AUTO DISABLED - fix me if you like :)
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    options
  )
}
