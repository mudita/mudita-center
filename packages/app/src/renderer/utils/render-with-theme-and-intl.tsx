/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "Renderer/locales/default/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"
import ReactModal from "react-modal"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
