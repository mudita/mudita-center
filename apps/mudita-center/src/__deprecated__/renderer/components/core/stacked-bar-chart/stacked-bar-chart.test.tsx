/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import React from "react"
import StackedBarChart, {
  DisplayStyle,
} from "App/__deprecated__/renderer/components/core/stacked-bar-chart/stacked-bar-chart.component"
import { renderWithThemeAndIntl } from "App/__deprecated__/renderer/utils/render-with-theme-and-intl"

const barId = "bar"
const barWithLabelId = "bar-with-label"

const chartData = [
  { value: 4294967296, color: "red" },
  { value: 4294967296, color: "orange" },
  { value: 4294967296, color: "yellow" },
  { value: 4294967296, color: "green" },
]
const occupiedSpaceId = "occupied-space"

describe("Snapshot tests", () => {
  test("should match snapshot simple", () => {
    const { container } = renderWithThemeAndIntl(
      <StackedBarChart chartData={chartData} displayStyle={DisplayStyle.Thin} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("should match snapshot multicolor", () => {
    const { container } = renderWithThemeAndIntl(
      <StackedBarChart
        chartData={chartData}
        displayStyle={DisplayStyle.Thick}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe("Correct rendering of chart elements and label", () => {
  test("without labels provided, bar with label should not be rendered", () => {
    const { queryByTestId } = renderWithThemeAndIntl(
      <StackedBarChart chartData={chartData} displayStyle={DisplayStyle.Thin} />
    )
    expect(queryByTestId(barWithLabelId)).not.toBeInTheDocument()
  })

  test("label should render", () => {
    const { getByTestId } = renderWithThemeAndIntl(
      <StackedBarChart
        chartData={chartData}
        displayStyle={DisplayStyle.Thick}
        labels
      />
    )
    expect(getByTestId(occupiedSpaceId)).toBeInTheDocument()
    expect(getByTestId(occupiedSpaceId)).toHaveTextContent(`17.2 GB( 100%)`)
  })

  test("number of elements rendered without label should equal to number of all elements - 1 (with label)", () => {
    const { getAllByTestId } = renderWithThemeAndIntl(
      <StackedBarChart
        chartData={chartData}
        displayStyle={DisplayStyle.Thick}
        labels
      />
    )
    expect(getAllByTestId(barId).length).toEqual(chartData.length - 1)
  })
})
