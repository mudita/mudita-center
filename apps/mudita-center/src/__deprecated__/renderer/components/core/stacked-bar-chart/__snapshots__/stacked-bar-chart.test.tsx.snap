// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should match snapshot multicolor 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c2 {
  width: 25%;
  background-color: red;
  --radius: 0.4rem;
  height: 0.8rem;
}

.c2:first-child {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c2:last-child {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c3 {
  width: 25%;
  background-color: orange;
  --radius: 0.4rem;
  height: 0.8rem;
}

.c3:first-child {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c3:last-child {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c4 {
  width: 25%;
  background-color: yellow;
  --radius: 0.4rem;
  height: 0.8rem;
}

.c4:first-child {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c4:last-child {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c5 {
  width: 25%;
  background-color: green;
  --radius: 0.4rem;
  height: 0.8rem;
}

.c5:first-child {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c5:last-child {
  border-radius: 0 var(--radius) var(--radius) 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      color="red"
      data-testid="bar"
    />
    <div
      class="c3"
      color="orange"
      data-testid="bar"
    />
    <div
      class="c4"
      color="yellow"
      data-testid="bar"
    />
    <div
      class="c5"
      color="green"
      data-testid="bar"
    />
  </div>
</div>
`;

exports[`Snapshot tests should match snapshot simple 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c2 {
  width: 25%;
  background-color: red;
  --radius: 0.2rem;
  height: 0.3rem;
}

.c2:first-child {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c2:last-child {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c3 {
  width: 25%;
  background-color: orange;
  --radius: 0.2rem;
  height: 0.3rem;
}

.c3:first-child {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c3:last-child {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c4 {
  width: 25%;
  background-color: yellow;
  --radius: 0.2rem;
  height: 0.3rem;
}

.c4:first-child {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c4:last-child {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c5 {
  width: 25%;
  background-color: green;
  --radius: 0.2rem;
  height: 0.3rem;
}

.c5:first-child {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c5:last-child {
  border-radius: 0 var(--radius) var(--radius) 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      color="red"
      data-testid="bar"
    />
    <div
      class="c3"
      color="orange"
      data-testid="bar"
    />
    <div
      class="c4"
      color="yellow"
      data-testid="bar"
    />
    <div
      class="c5"
      color="green"
      data-testid="bar"
    />
  </div>
</div>
`;
